src/jdk.jlink/share/classes/jdk/tools/jlink/internal/JlinkTask.java

```diff
@@ -580,7 +608,29 @@ private static ImageHelper createImageProvider(JlinkConfiguration config,
                         targetPlatform.arch().byteOrder(), targetPlatform);
             }
         }
-        return new ImageHelper(cf, mods, targetPlatform, retainModulesPath, ignoreSigning);
+        return new ImageHelper(cf, mods, targetPlatform, retainModulesPath,
+                               ignoreSigning, config.useModulePath(),
+                               config.singleHop(), nonClassRes);
+    }
+
+    private static void collectNonClassResources(String modName,
+                                                 Map<String, List<String>> nonResEntries,
+                                                 Module jdkJlink) {
+        String resName = String.format(OTHER_RESOURCES_FILE, modName);
+        try {
+            InputStream inStream = jdkJlink.getResourceAsStream(resName);
```

Comment:
Sure this can be done. Makes the jimage a bit larger, but probably negligible as the total set of modules is usually small.

Reply:
It adds more empty files which I expect should be very small increase in size.   Since this is added by default, we can measure the difference and confirm.

Reply:
Incidentally it won't save anything for the default JDK build, since all modules include legal files which end up being tracked in those files. So I don't think adding resource files unconditionally matters in practise.